### PR TITLE
chore: improve `All` tab for search results

### PR DESCRIPTION
**Pull request by shelegdmitriy**
This PR introduces changes to `TypeAheadDropdown` component for `All` tab only.

### CODE REVIEW

File: src/Search/TypeAheadDropdown.jsx [hunk 1]
```diff
@@ -249,6 +249,130 @@ const DisplayResultsByFacet = ({ title, count, items }) => (
...
+          .slice(0, topmostCount)
+          .map((hit) => ({ ...hit, variant: "components", relevance: hit.searchPosition }))
+      : [];
+  const postsAndComments =
+    state.postsAndComments.hits?.length > 0
+      ? state.postsAndComments.hits
+          .slice(0, topmostCount)
+          .map((hit) => ({ ...hit, variant: "postsAndComments", relevance: hit.searchPosition }))
+      : [];
+  const allResults = [...profiles, ...apps, ...components, ...postsAndComments];
```
Comment (charleslavon) [inline src/Search/TypeAheadDropdown.jsx:277]:
```suggestion
  const allResults = [ ...apps, ...profiles, ...components, ...postsAndComments];
```

**Comment by shelegdmitriy** [comment]
@charleslavon what do you think about this PR? Maybe we should leave a full screen without changes and improve only `TypeAheadDropdown` component?

**Review by charleslavon — APPROVED** [review]

File: src/Search/TypeAheadDropdown.jsx [hunk 2]
```diff
@@ -5,9 +5,6 @@ const API_URL = props.apiUrl ?? `https://${APPLICATION_ID}-dsn.algolia.net/1/ind
 const INITIAL_PAGE = props.initialPage ?? 0;
 const facets = props.facets ?? ["All", "People", "Apps", "Components", "Posts"];
 const tab = props.tab ?? "All";
-const showHeader = props.showHeader ?? true;
-const showSearchBar = props.showSearchBar ?? true;
-const showPagination = props.showPagination ?? true;
```
Comment (shelegdmitriy) [inline src/Search/TypeAheadDropdown.jsx:10]: This is unused properties

File: src/Search/TypeAheadDropdown.jsx [hunk 3]
```diff
@@ -16,14 +13,6 @@ State.init({
...
   selectedTab: tab,
   facet: tab,
-  isFiltersPanelVisible: false,
-  numColumns: 3,
-  selectedTags: [],
-  searchResults: [], // Assuming search results are stored here
-  allTags: [],
-  activeTags: [],
-  showFollowed: false,
-  showNotFollowed: false,
```
Comment (shelegdmitriy) [inline src/Search/TypeAheadDropdown.jsx:26]: This is unused properties

**Pull request merged**